spec/controllers/advertisements_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+include RandomData
+
+RSpec.describe AdvertisementsController, type: :controller do
+  let (:my_advertisement) do
+    Advertisement.create(
+      id: 1,
+      title: RandomData.random_sentence,
+      copy: RandomData.random_paragraph,
+      price: 99
+    )
+  end
+
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
+
+    it "assigns [my_advertisement] to @advertisements" do
+      get :index
+      expect(assigns(:advertisements)).to eq([my_advertisement])
+    end
+  end
+
+  describe "GET #show" do
+    it "returns http success" do
+      get :show, {id: my_advertisement.id}
+      expect(response).to have_http_status(:success)
+    end
+
+    it "renders the #show view" do
+      get :show, {id: my_advertisement.id}
+      expect(response).to render_template :show
+    end
+
+    it "assigns my_advertisement to @advertisement" do
+      get :show, {id: my_advertisement.id}
+      expect(assigns(:advertisement)).to eq(my_advertisement)
+    end
+  end
+
+  describe "GET new" do
+    it "returns http success" do
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+
+    it "renders the #new view" do
+      get :new
+      expect(response).to render_template :new
+    end
+
+    it "instantiates @advertisement" do
+      get :new
+      expect(assigns(:advertisement)).not_to be_nil
+    end
+  end
+
+  describe "POST create" do
+
+    it "increases the number of advertisements by 1" do
+      expect{post :create, advertisement: {title: RandomData.random_sentence, copy: RandomData.random_paragraph, price: 99}}.to change(Advertisement,:count).by(1)
```

Comment:
this line is a bit busy and hard to read.  consider move the hash of parameters into a `let` block.

Also, `change` should take a block, not arguments